test/views/ImportVATViewSpec.scala

```diff
@@ -0,0 +1,182 @@
+    }
+
+    "an error exists (no value has been specified for amended amount)" should {
+      lazy val form: Form[UnderpaymentAmount] = underpaymentFormWithValues(fifty, emptyString)
+      lazy val view: Html = injectedView(
+        form,
+        Call("GET", controllers.routes.UnderpaymentTypeController.onLoad().toString)
+      )(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      s"have the correct page title" in {
+        document.title mustBe ImportVATMessages.errorPrefix + ImportVATMessages.pageTitle
+      }
+
+      "render an error summary with the correct message " in {
+        elementText(govErrorSummaryListClass) mustBe ImportVATMessages.amendedNonEmpty
+      }
+
+      "render an error message against the field" in {
+        elementText(amendedErrorId) mustBe ImportVATMessages.errorPrefix + ImportVATMessages.amendedNonEmpty
+      }
+    }
+
+    "an error exists (not a numeric value has been specified for amended amount)" should {
+      lazy val form: Form[UnderpaymentAmount] = underpaymentFormWithValues(fifty, nonNumericInput)
+      lazy val view: Html = injectedView(
+        form,
+        Call("GET", controllers.routes.UnderpaymentTypeController.onLoad().toString)
+      )(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      s"have the correct page title" in {
+        document.title mustBe ImportVATMessages.errorPrefix + ImportVATMessages.pageTitle
+      }
+
+      "render an error summary with the correct message " in {
+        elementText(govErrorSummaryListClass) mustBe ImportVATMessages.amendedNonNumber
+      }
+
+      "render an error message against the field" in {
+        elementText(amendedErrorId) mustBe ImportVATMessages.errorPrefix + ImportVATMessages.amendedNonNumber
+      }
+    }
+
+    "an error exists (the value for original amount exceeds the limit)" should {
+      lazy val form: Form[UnderpaymentAmount] = underpaymentFormWithValues("10000000000", fifty)
+      lazy val view: Html = injectedView(
+        form,
+        Call("GET", controllers.routes.UnderpaymentTypeController.onLoad().toString)
+      )(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      s"have the correct page title" in {
+        document.title mustBe ImportVATMessages.errorPrefix + ImportVATMessages.pageTitle
+      }
+
+      "render an error summary with the correct message " in {
+        elementText(govErrorSummaryListClass) mustBe ImportVATMessages.originalUpperLimit
+      }
+
+      "render an error message against the field" in {
+        elementText(originalErrorId) mustBe ImportVATMessages.errorPrefix + ImportVATMessages.originalUpperLimit
+      }
+    }
+
+    }
+
+  it should {
+
+    lazy val form: Form[UnderpaymentAmount] = formProvider()
+    lazy val view: Html = injectedView(form,
+      Call("GET", controllers.routes.EntryDetailsController.onLoad().toString)
```

Comment:
BackLink data isn't valid for this page. It doesn't affect the test but not correct.

Reply:
changed to underpayment type controller